### PR TITLE
Remove public param and simplify verify param

### LINKED ISSUE #19: VDAFs that use the public parameter?
The "setup" algorithm outputs a public parameter used by the clients to shard their measurement into input shares. This was added to the syntax in anticipation of VDAFs that might make use of public-key cryptography in some way. For example, in the appendix, [BBCGGI19] describes alternative applications for FLPs, e.g., using homomorphic encryption. One can imagine a VDAF that uses homomorphic encryption in order to save a round trip for input validation. (This is basically what [AdScale](https://isi.jhu.edu/~mgreen/advertising.pdf) does, as I understand.)

This is a bit of a pain for PPM, since it means that rotating the verification parameter also requires re-configuring the clients with the new public parameter. (See https://github.com/abetterinternet/ppm-specification/issues/161.) Since we don't yet have a VDAF that makes use of the public parameter, it might be preferable to drop it from the syntax.

### PR DESCRIPTION
**Pull request by cjpatton**
Closes #19.

There are currently no VDAFs that use the public parameter. As noted
noted in https://github.com/cfrg/draft-irtf-cfrg-vdaf/issues/18 and https://github.com/cfrg/draft-irtf-cfrg-vdaf/issues/19, this creates friction for applications like DAP
(formerly known as PPM) that require frequent rotation of the
verification parameter shared by the Aggregators. This change addresses
this issue by dropping the public parameter from the syntax.

For our current set of VDAFs (Prio3 and Poplar1), the verification
parameter held by an Aggregator is basically a shared secret and some
sort of identifier of its role (aggregator 0, aggregator 1, ...). This
change codifies these two values in the syntax by dropping the
verification parameter and replacing it with a shared "verification key"
and an "aggregator ID". This allows us to replace the `setup()`
procedure with simply generating a shared secret, paving the way for a
solution for https://github.com/cfrg/draft-irtf-cfrg-vdaf/issues/18 that doesn't require breaking the abstraction boundary.

### CODE REVIEW

**Comment by schoppmp** [comment]
There are still a couple of `verify_param` throughout the document, for example under Security Considerations. Do you plan to replace them in this PR?

**Comment by schoppmp** [comment]
There are still a couple of `verify_param` throughout the document, for example under Security Considerations. Do you plan to replace them in this PR?

**Comment by cjpatton** [comment]
> There are still a couple of `verify_param` throughout the document, for example under Security Considerations. Do you plan to replace them in this PR?

I intended to leave the Poplar1 section unchanged, but missed the updates in Security Considerations. Nice catch, fixed.

File: draft-irtf-cfrg-vdaf.md [hunk 1]
```diff
@@ -734,14 +705,26 @@ share or an error."}
...
+  preparation state.
+
+  The length of `verify_key` MUST be `VERIFY_KEY_SIZE`. It is up to the
+  application to arrange for the distribution of the verification key among the
+  Aggregators prior to the start of this phase of VDAF evaluation.
+
+  > OPEN ISSUE: What security properties do we need for this key exchange? See
+  > issue#18.
+
+  The applicaiton MUST ensure that the Aggregator's identifier is equal to the
```
[the remark this answers is not in the document]
Comment (cjpatton) [reply draft-irtf-cfrg-vdaf.md]: Done.

**Review by schoppmp — APPROVED** [review]

File: draft-irtf-cfrg-vdaf.md [hunk 2]
```diff
@@ -626,54 +626,25 @@ listed in {{vdaf-param}} are defined by each concrete VDAF.
 
 | Parameter          | Description              |
 |:-------------------|:-------------------------|
+| `VERIFY_KEY_SIZE` | Size (in bytes) of the verification key ({{sec-vdaf-prepare}}) |
```
Comment (bifurcation) [inline draft-irtf-cfrg-vdaf.md:629]: Nit: Please add spaces below so that the column dividers align.

**Review by bifurcation — APPROVED** [review]

**Pull request merged**